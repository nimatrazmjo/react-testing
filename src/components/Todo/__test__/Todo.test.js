import { fireEvent, render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Todo from '../Todo';

const MockTodo = () => {
  return (
    <BrowserRouter>
      <Todo />
    </BrowserRouter>
  )
}

const addTodo = (tods) => {
  tods.forEach(todo => {
    const inputElement = screen.getByPlaceholderText(/Add a new task here.../i);
    const buttonElement = screen.getByText(/Add/i);
    fireEvent.change(inputElement, { target: { value: todo } });
    fireEvent.click(buttonElement);
  });
}

describe('Todo', () => { 
  it('Should render same text passed into title prop', () => {
    const task = 'Add new task';
    const { getByText, getByPlaceholderText } = render(<MockTodo />);
    addTodo([task])
    const devElement = screen.getByText(task);
    expect(devElement).toBeInTheDocument();
  });

  it('Should render multiple elements', () => {
    const { getByText, getByPlaceholderText } = render(<MockTodo />);
    const inputElement = getByPlaceholderText(/Add a new task here.../i);
    const buttonElement = getByText(/Add/i);
    fireEvent.change(inputElement, { target: { value: 'Add new task' } });
    fireEvent.click(buttonElement);
    
    fireEvent.change(inputElement, { target: { value: 'Add new task' } });
    fireEvent.click(buttonElement);
    fireEvent.change(inputElement, { target: { value: 'Add new task' } });
    fireEvent.click(buttonElement);
    const devElement = screen.getAllByTestId('task-container');
    expect(devElement).toHaveLength(3);
  });

  it('Tasks should not have completed class when initially rendered', () => {
    const { getByText, getByPlaceholderText } = render(<MockTodo />);
    const inputElement = getByPlaceholderText(/Add a new task here.../i);
    const buttonElement = getByText(/Add/i);
    fireEvent.change(inputElement, { target: { value: 'Add new task' } });
    fireEvent.click(buttonElement);
    const devElement = screen.getByTestId('task-container');
    expect(devElement).not.toHaveClass('todo-item-active');
  });

  it('Tasks should have completed class when completed', () => {
    const { getByText, getByPlaceholderText } = render(<MockTodo />);
    const inputElement = getByPlaceholderText(/Add a new task here.../i);
    const buttonElement = getByText(/Add/i);
    fireEvent.change(inputElement, { target: { value: 'Add new task' } });
    fireEvent.click(buttonElement);
    const devElement = screen.getByTestId('task-container');
    fireEvent.click(devElement);
    expect(devElement).toHaveClass('todo-item-active');
  });
});
