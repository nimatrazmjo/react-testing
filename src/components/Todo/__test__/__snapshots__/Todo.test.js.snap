// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo renders correctly 1`] = `
<div>
  <div
    class="todo"
  >
    <h1
      class="header"
      data-testid="header-1"
    >
      Todo
    </h1>
    <h3
      class="header"
    >
      nimat razmjo
    </h3>
    <div
      class="input-container"
    >
      <input
        class="input"
        placeholder="Add a new task here..."
        value=""
      />
      <button
        class="add-btn"
      >
        Add
      </button>
    </div>
    <div
      class="todolist-container"
    >
      <div
        class="todos-container"
      >
        <div />
      </div>
      <div>
        <div
          class="todo-footer"
        >
          <p>
            0
             
            tasks
             left
          </p>
          <a
            href="/followers"
          >
            Followers
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
